server/controllers/Imap-connection.js

```diff
@@ -1,44 +1,41 @@
 const Imap = require('imap');
 const { MailParser } = require('mailparser-mit');
+const events = require('../controllers/socket');
 require('dotenv').config();
 
-const imap = new Imap({
-  user: process.env.IMAP_USER,
-  password: process.env.IMAP_USER_PASS,
-  host: 'imap.gmail.com',
-  port: 993,
-  tls: true,
-});
 
 const mails = (
-  triggerGetMailsObj,
-  triggerOnNewMail,
-  triggerUpdateStatusObj,
-  triggerSearchKeyword,
   socket,
   io,
-  getMailsCallback,
 ) => {
+  const imap = new Imap({
+    user: process.env.IMAP_USER,
+    password: process.env.IMAP_USER_PASS,
+    host: 'imap.gmail.com',
+    port: 993,
+    tls: true,
+  });
+
   imap.once('ready', () => {
-    if (triggerGetMailsObj) {
-      imap.openBox('INBOX', false, (err, box) => {
-        const f = imap.seq.fetch(triggerGetMailsObj.range, { bodies: '' });
+    imap.openBox('INBOX', false, (err, box) => {
```

Comment:
callback hell.

Try to avoid using callbacks and wrap any functions you can in promises.